drivers/modem/hl7800.c

```diff
@@ -0,0 +1,4900 @@
+	char mdm_iccid[MDM_HL7800_ICCID_SIZE];
+	uint8_t mdm_startup_state;
+	enum mdm_hl7800_radio_mode mdm_rat;
+	char mdm_active_bands_string[MDM_HL7800_LTE_BAND_STR_SIZE];
+	char mdm_bands_string[MDM_HL7800_LTE_BAND_STR_SIZE];
+	uint16_t mdm_bands_top;
+	uint32_t mdm_bands_middle;
+	uint32_t mdm_bands_bottom;
+	int32_t mdm_sinr;
+	bool mdm_echo_is_on;
+	struct mdm_hl7800_apn mdm_apn;
+	bool mdm_startup_reporting_on;
+	int device_services_ind;
+
+	/* modem state */
+	bool allowSleep;
+	bool uartOn;
+	enum mdm_hl7800_sleep_state sleep_state;
+	enum mdm_hl7800_network_state network_state;
+	enum net_operator_status operator_status;
+	void (*event_callback)(enum mdm_hl7800_event event, void *event_data);
+#ifdef CONFIG_NEWLIB_LIBC
+	struct tm local_time;
+	int32_t local_time_offset;
+#endif
+	bool local_time_valid;
+	bool configured;
+};
+
+struct cmd_handler {
+	const char *cmd;
+	uint16_t cmd_len;
+	bool (*func)(struct net_buf **buf, uint16_t len);
+};
+
+static struct hl7800_iface_ctx ictx;
+
+static size_t hl7800_read_rx(struct net_buf **buf);
+static char *get_network_state_string(enum mdm_hl7800_network_state state);
+static char *get_startup_state_string(enum mdm_hl7800_startup_state state);
+static char *get_sleep_state_string(enum mdm_hl7800_sleep_state state);
+static void set_network_state(enum mdm_hl7800_network_state state);
+static void set_startup_state(enum mdm_hl7800_startup_state state);
+static void set_sleep_state(enum mdm_hl7800_sleep_state state);
+static void generate_network_state_event(void);
+static void generate_startup_state_event(void);
+static void generate_sleep_state_event(void);
+static int modem_boot_handler(char *reason);
+static void mdm_vgpio_work_cb(struct k_work *item);
+static void mdm_reset_work_callback(struct k_work *item);
+static int write_apn(char *access_point_name);
+
+#ifdef CONFIG_MODEM_HL7800_FW_UPDATE
+static char *get_fota_state_string(enum mdm_hl7800_fota_state state);
+static void set_fota_state(enum mdm_hl7800_fota_state state);
+static void generate_fota_state_event(void);
+static void generate_fota_count_event(void);
+#endif
+
+#ifdef CONFIG_NEWLIB_LIBC
+static bool convert_time_string_to_struct(struct tm *tm, int32_t *offset,
+					  char *time_string);
+#endif
+
+#ifdef CONFIG_MODEM_HL7800_LOW_POWER_MODE
+static bool is_cmd_ready(void)
+{
+	ictx.vgpio_state = (uint32_t)gpio_pin_get(ictx.gpio_port_dev[MDM_VGPIO],
+						  pinconfig[MDM_VGPIO].pin);
+
+	ictx.gpio6_state = (uint32_t)gpio_pin_get(ictx.gpio_port_dev[MDM_GPIO6],
+						  pinconfig[MDM_GPIO6].pin);
+
+	ictx.cts_state = (uint32_t)gpio_pin_get(
+		ictx.gpio_port_dev[MDM_UART_CTS], pinconfig[MDM_UART_CTS].pin);
+
+	return ictx.vgpio_state && ictx.gpio6_state && !ictx.cts_state;
+}
+#endif
+
+/**
+ * The definition of awake is that the HL7800
+ * is ready to receive AT commands successfully
+ */
+static void check_hl7800_awake(void)
+{
+#ifdef CONFIG_MODEM_HL7800_LOW_POWER_MODE
+	bool isCmdRdy = is_cmd_ready();
```

Comment:
camelCase is not recommended for Zephyr code, snake case should be used instead.

Reply:
> camelCase is not recommended for Zephyr code, snake case should be used instead.

Can you point me to this recommendation in any of the zephyr documentation, I couldn't find anything. In the interest of productivity, I plan to leave this as is unless it is deemed critical that I change all of it.

Reply:
From https://www.kernel.org/doc/html/latest/process/coding-style.html, which we follow:
> HOWEVER, while **mixed-case names are frowned upon**, descriptive names for global variables are a must.

As I said, it's not a hard requirement, but a recommendation. Personally, I find it a bit messy, that there are places where even in a single function variables names are following different styles.

Reply:
I missed these camel-case vars in my review, but I agree with @rlubos here, it would be much easier to read/maintain the code if consistent style is followed in the code base. I already gave +1, but @rlubos I suggest that you just give -1 if/when needed or see relevant.